test/functional/treesitter/highlight_spec.lua

```diff
@@ -570,4 +570,47 @@ describe('treesitter highlighting', function()
     ]]}
     screen:expect{ unchanged=true }
   end)
+
+  it("supports highlighting with priority", function()
+    if pending_c_parser(pending) then return end
+
+    insert([[
+    int x = INT_MAX;
+    #define READ_STRING(x, y) (char_u *)read_string((x), (size_t)(y))
+    #define foo void main() { \
+                  return 42;  \
+                }
+    ]])
+
+    exec_lua [[
+      local parser = vim.treesitter.get_parser(0, "c")
+      test_hl = vim.treesitter.highlighter.new(parser, {queries = {c = hl_query..'\n((translation_unit) @Error (set! "priority" 101))\n'}})
+    ]]
+    -- expect everything to have Error highlight
+    screen:expect{grid=[[
+      {12:int}{8: x = INT_MAX;}                                                 |
```

Comment:
Looks like we are keeping the two highlights here (the bold is kept), so I am wondering if it is what we really want to do?

Reply:
That would be for another pr though.